llarp/config/config.cpp

```diff
@@ -834,48 +834,66 @@ namespace llarp
   }
 
   LinksConfig::LinkInfo
-  LinksConfig::LinkInfoFromINIValues(std::string_view name, std::string_view value)
+  LinksConfig::LinkInfoFromINIValues(
+      std::string_view name, std::string_view value, const net::Platform& net)
   {
     // we treat the INI k:v pair as:
-    // k: interface name, * indicating outbound
+    // k: interface name, * indicating outbound, inbound for inbound addr, outbound for outbound
     // v: a comma-separated list of values, an int indicating port (everything else ignored)
-    //    this is somewhat of a backwards- and forwards-compatibility thing
-
-    LinkInfo info;
-    info.port = 0;
-    info.addressFamily = AF_INET;
-
-    if (name == "address")
+    //    this is somewhat of a backwards- and forwards-compatibility thing, or if k is outbound,
+    //    inbound or *
+    if (name == "inbound" or name == "outbound")
     {
-      const IpAddress addr{value};
-      if (not addr.hasPort())
-        throw std::invalid_argument("no port provided in link address");
-      info.m_interface = addr.toHost();
-      info.port = *addr.getPort();
+      const SockAddr addr{value};
+      LinkInfo info{};
+
+      if (net.IsLoopbackAddress(addr.getIP()))
+        throw llarp::make_exception<std::invalid_argument>(
+            value, " is a loopback address and cannot be used with link addresses");
+
+      info.m_interface = var::visit([](auto&& ip) { return ip.ToString(); }, addr.getIP());
```

Comment:
It feels strange to set something that is always an IP address into a variable named `m_interface`.

Reply:
it really should be a `llarp::SockAddr` at this point instead of string and int